test/uk/gov/hmrc/customsexportsinternalfrontend/controllers/HelloWorldControllerSpec.scala

```diff
@@ -16,40 +16,31 @@
 
 package uk.gov.hmrc.customsexportsinternalfrontend.controllers
 
-import org.scalatest.{Matchers, WordSpec}
-import org.scalatestplus.play.guice.GuiceOneAppPerSuite
 import play.api.http.Status
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
-import play.api.{Configuration, Environment, _}
-import uk.gov.hmrc.play.bootstrap.config.{RunMode, ServicesConfig}
-import uk.gov.hmrc.play.bootstrap.tools.Stubs.stubMessagesControllerComponents
-import uk.gov.hmrc.customsexportsinternalfrontend.config.AppConfig
+import uk.gov.hmrc.customsexportsinternalfrontend.controllers.actions.{AuthenticatedAction, Verify}
 import uk.gov.hmrc.customsexportsinternalfrontend.views.html.hello_world
+import uk.gov.hmrc.play.bootstrap.tools.Stubs.stubMessagesControllerComponents
 
-class HelloWorldControllerSpec extends WordSpec with Matchers with GuiceOneAppPerSuite {
-  private val fakeRequest = FakeRequest("GET", "/")
-
-  private val env = Environment.simple()
-  private val configuration = Configuration.load(env)
+class HelloWorldControllerSpec extends ControllerLayerSpec {
 
-  private val serviceConfig = new ServicesConfig(configuration, new RunMode(configuration, Mode.Dev))
-  private val appConfig = new AppConfig(configuration, serviceConfig)
-  private val helloWorldPage = app.injector.instanceOf[hello_world]
+  private val helloWorldPage = aRealInstanceOf[hello_world]
```

Comment:
Why we test view and controller toggether? Looks like mix of unit and interaction tests.

Reply:
Calling view is what controller does - you can use mock here. Counter prof  by [example](https://github.com/hmrc/customs-declare-exports-frontend/blob/91a50967a6eecf9b7d5a9281a89cc149d855231a/test/unit/controllers/declaration/PreviousDocumentsControllerSpec.scala#L37)